### PR TITLE
feat: implement get user shows service

### PR DESCRIPTION
**Pull request by okanisildar**
feat: implement get user shows service

chore: change types of search user shows function return

### CODE REVIEW

**Comment by okanisildar** [comment]
Current dependencies on/for this PR:
* main
  * **PR #24** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/24?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #25** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/25?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/experiment-station/beecast/24?utm_source=stack-comment).

**Review by altaywtf — APPROVED** [review]

File: lib/services/spotify/get-user-shows.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+type SpotifyItems = {
+  items: Items[];
+  limit: number;
+  offset: number;
+};
+
+export const getUserShows = async (spotifyToken: string): Promise<Items[]> => {
+  try {
+    const response = await fetch('https://api.spotify.com/v1/me/shows', {
```
[the remark this answers is not in the document]
Comment (altaywtf) [reply lib/services/spotify/get-user-shows.ts:24]: also supports creating `clients` to interact with sdks like spotify:

```
const spotifyFetchClient = ofetch.create({
    baseURL: 'https://api.spotify.com/v1',
    headers: {
      Authorization: `Bearer ${spotifyToken}`,
    },
  });
```

[the remark this answers is not in the document]
Comment (altaywtf) [reply lib/services/spotify/get-user-shows.ts:24]: also this :D https://github.com/unjs/ofetch?tab=readme-ov-file#%EF%B8%8F-handling-errors

**Review by altaywtf — APPROVED** [review]
solid 👍

**Pull request merged**